youtube_dlc/extractor/niconico.py

```diff
@@ -284,80 +360,132 @@ def _real_extract(self, url):
+        )
+
+        timestamp = (
+            parse_iso8601(get_video_info_web('first_retrieve'))
+            or unified_timestamp(get_video_info_web('postedDateTime'))
+        )
+
+        # According to my personal research,
+        # 'creation_time' tag on video stream of re-encoded SMILEVIDEO mp4 files are '1970-01-01T00:00:00.000000Z'.
+        metadata_timestamp = (
+            parse_iso8601(try_get(v_stream, lambda x: x['tags']['creation_time']))
+            or timestamp
+        )
+
+        # According to compconf and my personal research, smile videos from pre-2017 are always better quality than their DMC counterparts
+        smile_threshold_timestamp = parse_iso8601('2016-12-08T00:00:00+09:00')
+
+        is_source = timestamp < smile_threshold_timestamp or metadata_timestamp > 0
+
+        # If movie file size is unstable, old server movie is not source movie.
+        if int(get_video_info_xml('size_high')) > 1:
+            formats.append({
                 'url': video_real_url,
+                'format_id': 'smile' if not is_economy else 'smile_low',
+                'format_note': 'SMILEVIDEO source' if not is_economy else 'SMILEVIDEO low quality',
                 'ext': extension,
-                'format_id': _format_id_from_url(video_real_url),
-            }]
-        else:
-            formats = []
-
-            dmc_info = api_data['video'].get('dmcInfo')
-            if dmc_info:  # "New" HTML5 videos
-                quality_info = dmc_info['quality']
-                for audio_quality in quality_info['audios']:
-                    for video_quality in quality_info['videos']:
-                        if not audio_quality['available'] or not video_quality['available']:
-                            continue
-                        formats.append(self._extract_format_for_quality(
-                            api_data, video_id, audio_quality, video_quality))
-
-                self._sort_formats(formats)
-            else:  # "Old" HTML5 videos
-                formats = [{
-                    'url': video_real_url,
-                    'ext': 'mp4',
-                    'format_id': _format_id_from_url(video_real_url),
-                }]
-
-            def get_video_info(items):
-                return dict_get(api_data['video'], items)
+                'container': extension,
+                'vcodec': dict_get(v_stream, 'codec_name'),
+                'acodec': dict_get(a_stream, 'codec_name'),
+                # Some complex swf files doesn't have total bit rate metadata (e.g. nm6049209)
+                'tbr': int_or_none(metadata['format'].get('bit_rate', None), scale=1000),
+                'vbr': int_or_none(v_stream.get('bit_rate', None), scale=1000),
+                'abr': int_or_none(a_stream.get('bit_rate', None), scale=1000),
+                'height': int_or_none(v_stream.get('height', None)),
+                'width': int_or_none(v_stream.get('width', None)),
```

Comment:
```suggestion
                'tbr': int_or_none(metadata['format'].get('bit_rate'), scale=1000),
                'vbr': int_or_none(v_stream.get('bit_rate'), scale=1000),
                'abr': int_or_none(a_stream.get('bit_rate'), scale=1000),
                'height': int_or_none(v_stream.get('height')),
                'width': int_or_none(v_stream.get('width')),
```

`None` is the default for `.get`